_posts/en/posts/2016-01-25-segwit-benefits.md

```diff
@@ -0,0 +1,208 @@
+
+## Linear scaling of sighash operations
+
+A major problem with simple approaches to increasing the bitcoin blocksize
+is that for certain transactions, signature-hashing scales quadratically
+rather than linearly -- in essence, doubling the size of a transaction
+increases can double both the number of signature operations, and the
+amount of data that has to be hashed for each of those signatures to
+be verified. This has been seen in the wild, where an individual block
+required 25 seconds to validate, and maliciously designed transactions
+could take over 3 minutes.
+
+Segwit resolves this by changing the calculation of the transaction hash
+for signatures so that each byte of a transaction only needs to be hashed
+at most twice. This provides the same functionality more efficiently,
+so that large transactions can still be generated without running into
+problems due to signature hashing, even if they are generated maliciously,
+or larged transactions are supported.
+
+#### Who benefits?
+
+Removing the quadratic scaling of hashed data for verifying signatures
+makes increasing the block size safer. Doing that without also limiting
+transaction sizes allows bitcoin to continue to support payments to and
+from large groups, such as payments of mining rewards or crowdfunding
+services.
+
+#### Further information
+
+ * [BIP 143](https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki)
+ * [Blog post by Rusty Russell on the 25s transaction](http://rusty.ozlabs.org/?p=522)
+
+## Signing of input values
+
+When a hardware wallet signs a transaction, it can easily verify the total
+amount being spent, but can only safely determine the fee by having a
+full copy of all the input transactions being spent, and must hash each
+of those to ensure it is not being fed false data. Since individual
+transactions can be up to 1MB in size, this is not necessarily a cheap
+operation, even if the transaction being signed is itself quite small.
+
+Segwit resolves this by explicitly hashing the input value. This means
+that a hardware wallet can simply be given the transaction hash, index,
+and value (and told what public key was used), and can safely sign the
+spending transaction, no matter how large or complicated the transaction
+being spent was.
+
+#### Who benefits?
+
+Manufactures and users of hardware wallets are the obvious beneficiaries;
+however this likely also makes it much easier to safely use bitcoin in
+small embedded devices for "Internet of things" applications.
+
+#### Further information
+
+ * [BIP 143](https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki)
+
+## Increased security for multisig via Pay-to-script-hash
+
+Multisig payments currently use P2SH which is secured by the 160-bit
+HASH160 algorithm (RIPEMD of SHA256). However, a collision can be found
+with only 80-bits (2<sup>80</sup>) worth of work, which is already within
+the realm of possibility for an extremely well-resourced attacker.
```

Comment:
Perhaps mention that Bitcoin mining currently does the necessary amount of work for that in less than a month?